server/src/repositories/config.repository.ts

```diff
@@ -69,12 +68,21 @@ const getEnv = (): EnvData => {
     }
   }
 
-  const globalEnabled = parseBoolean(process.env.IMMICH_METRICS, false);
-  const hostMetrics = parseBoolean(process.env.IMMICH_HOST_METRICS, globalEnabled);
-  const apiMetrics = parseBoolean(process.env.IMMICH_API_METRICS, globalEnabled);
-  const repoMetrics = parseBoolean(process.env.IMMICH_IO_METRICS, globalEnabled);
-  const jobMetrics = parseBoolean(process.env.IMMICH_JOB_METRICS, globalEnabled);
-  const telemetryEnabled = globalEnabled || hostMetrics || apiMetrics || repoMetrics || jobMetrics;
+  const isAll = (telemetries: Set<ImmichTelemetry | 'all'>): telemetries is Set<'all'> =>
+    telemetries.has('all') && telemetries.size === 1;
+
+  let includedTelemetries = asSet<ImmichTelemetry | 'all'>(process.env.IMMICH_TELEMETRY_INCLUDE, []);
+  if (isAll(includedTelemetries)) {
+    includedTelemetries = new Set(Object.values(ImmichTelemetry)) as Set<ImmichTelemetry>;
+  }
+
+  const excludedTelemetries = asSet<ImmichTelemetry>(process.env.IMMICH_TELEMETRY_EXCLUDE, []);
+  const telemetries = setDifference(includedTelemetries as Set<ImmichTelemetry>, excludedTelemetries);
```

Comment:
```suggestion
  const includedTelemetries =
    process.env.IMMICH_TELEMETRY_INCLUDE === 'all'
      ? asSet<ImmichTelemetry>(process.env.IMMICH_TELEMETRY_INCLUDE, [])
      : new Set(Object.values(ImmichTelemetry));

  const excludedTelemetries = asSet<ImmichTelemetry>(process.env.IMMICH_TELEMETRY_EXCLUDE, []);
  const telemetries = setDifference(includedTelemetries, excludedTelemetries);
```

Reply:
this is what I was thinking too @danieldietzler

Reply:
Ooooh that makes so much more sense lol